### PR TITLE
Merge tibble and pillar vignettes

### PR DESCRIPTION
**Pull request by krlmlr**
I forgot we alread had an "extending" vignette in tibble. I merged the existing and new vignettes.

@mvers81: Can you please review?

### CODE REVIEW

**Comment by krlmlr** [comment]
Rendered version: http://rpubs.com/krlmlr/extending-pillar

**Comment by mvers81** [comment]
Looks good.

**Comment by krlmlr** [comment]
@hadley: I forgot we already had such a vignette in tibble. Should we move the vignette to tibble, or keep it here and add a link in the tibble vignette?

File: vignettes/extending.Rmd [hunk 1]
```diff
@@ -9,29 +9,64 @@ vignette: >
...
+To extend the tibble package for new types of columnar data, you need to understand how printing works. The presentation of a column in a tibble is powered by four S3 generics:
+
+* `type_sum()` determines what goes into the column header.
+* `pillar_shaft()` determines what goes into the body of the column.
+* `is_vector_s3()` and `obj_sum()` are used when rendering list columns.
+
+If you have written an S3 or S4 class that can be used as a column, you can override these generics to make sure your data prints well in a tibble. To start, you must import the `pillar` package that powers the printing of tibbles. Either add `pillar` to the `Imports:` section of your `DESCRIPTION`, or simply call:
+
+```{r, eval = FALSE}
+devtools::use_package("pillar")
```
Comment (hadley) [inline vignettes/extending.Rmd]: Better to use usethis now

**Pull request merged**